test/jest/types.spec.ts

```diff
@@ -0,0 +1,593 @@
+        expect(PhoneNumber2).toEqual(20);
+        expect(PhoneNumber3).toEqual(21);
+    });
+
+    describe('#getValuesAsStrings', () => {
+        it('returns an array of identity types values as an array of numbers as strings', () => {
+            const { getValuesAsStrings } = IdentityType;
+
+            expect(getValuesAsStrings()).toEqual([
+                '0',
+                '1',
+                '2',
+                '3',
+                '4',
+                '5',
+                '6',
+                '7',
+                '9',
+                '10',
+                '11',
+                '12',
+                '13',
+                '14',
+                '15',
+                '16',
+                '17',
+                '18',
+                '19',
+                '20',
+                '21',
+            ]);
+        });
+    });
+
+    describe('#isValid', () => {
+        it('returns true if the identity type is valid', () => {
+            const { isValid } = IdentityType;
+
+            expect(isValid(Other)).toBe(true);
+            expect(isValid(CustomerId)).toBe(true);
+            expect(isValid(Facebook)).toBe(true);
+            expect(isValid(Twitter)).toBe(true);
+            expect(isValid(Google)).toBe(true);
+            expect(isValid(Microsoft)).toBe(true);
+            expect(isValid(Yahoo)).toBe(true);
+            expect(isValid(Email)).toBe(true);
+            expect(isValid(FacebookCustomAudienceId)).toBe(true);
+            expect(isValid(Other2)).toBe(true);
+            expect(isValid(Other3)).toBe(true);
+            expect(isValid(Other4)).toBe(true);
+            expect(isValid(Other5)).toBe(true);
+            expect(isValid(Other6)).toBe(true);
+            expect(isValid(Other7)).toBe(true);
+            expect(isValid(Other8)).toBe(true);
+            expect(isValid(Other9)).toBe(true);
+            expect(isValid(Other10)).toBe(true);
+            expect(isValid(MobileNumber)).toBe(true);
+            expect(isValid(PhoneNumber2)).toBe(true);
+            expect(isValid(PhoneNumber3)).toBe(true);
+        });
+
+        it('returns false if the identity type is not valid', () => {
+            const { isValid } = IdentityType;
+
+            expect(isValid(NaN)).toBe(false);
+            expect(isValid('invalid' as unknown as number)).toBe(false);
+        });
+    });
+
+    describe('#getNewIdentitesByName', () => {
+        it('returns an identity name when passing an identity type', () => {
+            const { getNewIdentitiesByName } = IdentityType;
+
+            const newIdentitiesByType = {
+                [CustomerId]: 'foo',
+                [Email]: 'bar@gmail.com',
+            };
+
+            expect(getNewIdentitiesByName(newIdentitiesByType)).toEqual({
+                customerid: 'foo',
+                email: 'bar@gmail.com',
+            });
+        });
+
+        it('returns an empty object if the identity type is not found', () => {
```

Comment:
I think this test name is a bit misleading.  If an object is full of identity types that aren't real, then an empty object is returned.  Really this test should perhaps have 1 valid type, and the test should be "removes invalid identity types..."